### PR TITLE
Issue #11214: Specify violation messages for UnnecessaryParenthesesCheck

### PR DESCRIPTION
**Pull request by faraazb**
Specified violation messages in input files for **one** of the checks - `UnnecessaryParenthesesCheck` - in Issue #11214 .

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (Vyom-Yadav) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:21]: The rationale behind specifying violation messages in input files was to make them easier to read and avoid traveling back to the test file to see what the actual violation is, `.* right-hand side` doesn't clearly imply what the violation message could be. Here the original message is-
`Unnecessary parentheses around assignment right-hand side.`
You can shorten it to-
`Unnecessary paren.* around assignment right.* side` or to anything else which implies what the actual violation is.

If the line length doesn't cooperate then you could shorten the variables name, wrap it on the next line or use violation above or below pattern (last resort)

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (Vyom-Yadav) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:20]: Same here, `.* expression` is not very descriptive.

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (Vyom-Yadav) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:21]: violation above/below pattern should be considered as the very last resort. The case which you have mentioned is a chained method call-
```java
new HashSet<Integer>().stream().filter((filter) -> filter > 0); // violation '.* lambda .*'
```
It is good to wrap chained method calls for readability, I would have done something like-
```java
new HashSet<Integer>()
    .stream()
    .filter((filter) -> filter > 0);  // violation .........
```
or at least would have wrapped `filter` onto the next line (as some prefer to keep `stream()` on the first line).
> will introduce a new line and I will have to make changes in the expected line numbers in the test, right? Can I do that?

Absolutely, you can modify the test file.

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:21]: Change indentation from +4 to +2, it will give extra space for better message

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:21]: Rename MathOperation2 to MathOp2 to give you even more space.
Anything that is not about parentheses are ok to change/shorten

**Comment by faraazb** [comment]
Thank you so much @Vyom-Yadav and @romani! I have made the changes.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (Vyom-Yadav) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:22]: No need for leading or trailing `.*` it gets added automatically.
`.* parentheses around assignment right.*side` 
↓
 `parentheses around assignment right.*side`
Please change it for all the lines.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java
Comment (Vyom-Yadav) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: This is not required, please revert this.

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Few Items-
Also please change the PR heading to `Issue #XXXX:` format i.e. same as the main commit message.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: I will revert it. I made this change to make space [here](https://github.com/faraazb/checkstyle/blob/db6a1205a592e10bcd14b1845b531d11b857581f/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java#L95). I guess 'violation above' is better than wrapping in this case?

Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: Over there it is fine, here renaming to `GenT` is not of any use, there renaming made more space, I understand you are trying to be consistent with naming ut it doesn't matter in the test file unless it's a naming related check.

**Comment by Vyom-Yadav** [comment]
Remove the leading `#` from PR desc and place it before issue number.

**Comment by romani** [comment]
@Vyom-Yadav, please make comment "resolved" in your review points to let me/maintainers actually resolve them and see that you are satisfied for all points. We appreciate code reviews from everyone, but GitHub permissions requires improvement to  self resolve reviews by author.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: Since this input file gets compiled I cannot rename (`TypeParameterized` to `GenT`) only on line 95. How should I go about?

**Comment by romani** [comment]
@faraazb, please reply "done" to each review item thread, as signal that you addressed such review item

**Comment by romani** [comment]
@faraazb, please reply "done" to each review item thread, as signal that you addressed such review item

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java]: Done

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (faraazb) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: Oh ok, you are using this class as ref var type over there, it's fine.

Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesOperatorsAndCasts.java:112]: Resolved.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (Vyom-Yadav) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java]: Resolved

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java]: Resolved

**Comment by romani** [comment]
@nick-mancuso , please share link to your GitHub request for improvement to allow resolving review item. @Vyom-Yadav , if you agree please vote on it

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:22]: @faraazb , ∆∆

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: >around assignment right-hand side

Assignment ??
Looks like we found a defect in our Check :), it is not assignment.
Please create issue on this.
This problem will not block this is PR, but Issue  need to be created.

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: @rnveach , @pbludov , @strkkk , @nick-mancuso 
Please pay attention to this, good example how good message in input file can help

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:21]: Please do me favor, remove content of this line, it was good hint when we had no support of messages of violation 
It should be empty line now

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:89]: Remove content from this line, it should be empty line.

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:89]: Is there a reason to keep it empty? Why can't we remove it?

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:89]: If you remove it, it will force you to shift all line number in test method expected array.
I just try to avoid extra work for you and extra changes for me to review

Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:89]: Thanks! But if it is okay with you, can I remove those two empty lines?

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:89]: Ok to remove

Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java:87]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java
Comment (faraazb) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java]: Done

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (faraazb) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java]: Done

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: @faraazb , please create issue

**Review by romani — APPROVED** [review]
Ok to merge

@Vyom-Yadav , please finish review

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (Vyom-Yadav) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java]: Resolved.

**Review by Vyom-Yadav — APPROVED** [review]
I updated the list of checks completed in the issue.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: Please share link to issue here

Comment (faraazb) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: Sorry for taking some time. I tried to understand what may have been causing it. I will create an issue and link it here, very shortly.

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: We can investigate later on, let's not block this PR.

Comment (faraazb) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: Please see #11357

**Comment by strkkk** [comment]
@faraazb lgtm, please rebase to latest master and resolve conflict

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java:36]: Thanks a lot for effort on issue creation

**Pull request merged**
